### DISCUSSION TITLE
Where can I find more information about the concept of relational_metadata?

### DISCUSSION THREAD

**Discussion by githubpiyush** [Sunbird-Knowlg/Community#48] [General Discussions]
We are given to understand there is concept of relational_metadata which is used for collection objectTypes and it kind of helps define/store relationship between two objectTypes. 

If someone can point us to relevant documentation about it and how it is presently being used in Sunbird-Knowlg. 

Context :- 

PRD link :: https://project-sunbird.atlassian.net/wiki/spaces/SingleSource/pages/2983002141/Design+-+Observation+with+Rubrics (The 3rd Approach)

story link :: https://project-sunbird.atlassian.net/browse/SB-27420

Proposed Idea :-
Add below metadata to the root questionset node of the observation-with-rubrics (primaryCategory).

```
questionCustomConfig:{
        do_5678 : { 			//Question Identifier
            ecm: CODE_111,
            textColor: blue
        }
},

ecms:[{
        code: CODE_111,
        value: 'Infra',
        canBeNotApplicable: 'true'
    },{
        code: CODE_222,
        value: 'Efficiency',
        canBeNotApplicable: 'false'
}]
```

Note : - These fields apply only to **Observation-With-Rubrics**, as such, we would like to store it as part of primary category definition, instead of questionSet schema.json.

But since these fields come into effect only as part of the relationship between questionSet and questions, Mahesh suggested that we should use relational_metadata for storing the above information. 

CC :- @aks30 @joffinjoy @kiranharidas187  @pradeepkumarcm96 @pallakartheekreddy @MaheshKulkarniTekdi
 @vinukumar-vs @AmiableAnil  @krgauraw

**Comment by Jayaprakash8887** [comment]
https://project-sunbird.atlassian.net/wiki/spaces/SingleSource/pages/1977745409/Generalized+Collection+Editor

**Comment by Jayaprakash8887** [comment]
https://project-sunbird.atlassian.net/wiki/spaces/SBDES/pages/3089137672/Relational+Metadata+feature+Release+Notes

**Comment by Jayaprakash8887** [comment]
https://project-sunbird.atlassian.net/wiki/spaces/SBDES/pages/3015770158/Design+-+Relational+metadata+to+asset+linked+to+a+collection?atlOrigin=eyJpIjoiMmUxNWEwMGNhNjU4NGQ4NGFiZjVlMTZjYzAzNTUyYzQiLCJwIjoiY29uZmx1ZW5jZS1jaGF0cy1pbnQifQ

**Comment by githubpiyush** [comment]
@pallakartheekreddy @MaheshKulkarniTekdi @vinukumar-vs @AmiableAnil  @krgauraw

Help us with these points:-
1. Please update us on whether relational_metadata is enabled for Question and QuestionSet APIs. (meeting MOM Action Item 3)
2. As per our understanding relationship is linked between children(Question) and immediate parent(QuestionSet) but is it possible to have relationship between childrens(Question which are not the intermediate nodes) and root node(QuestionSet).
3. Can we store the ECM information for every question in the questionset's root node? If yes please give us a hierarchy read api response with relational metadata example as we didn't find any similar example in any of the above links.

MOM link :- https://docs.google.com/document/d/1maVfT0qykU57OnaHCecphp630QPnbaRde4zdzR7pxCY/edit?pli=1#

CC :- @aks30 @joffinjoy @kiranharidas187 @pradeepkumarcm96

**Comment by Jayaprakash8887** [comment]
@vinukumar-vs We have had a call with the team to discuss about relational_metadata feature was developed for collection and its resources; and how the same approach could be followed by them to develop the feature for Question and QuestionSet
